scripts/zkai_native_seq32_attention_mlp_dry_run_opening_sampler_gate.py

```
@@ -0,0 +1,778 @@
+
+
+def atomic_write_pair(
+    json_path: pathlib.Path,
+    json_data: bytes,
+    tsv_path: pathlib.Path,
+    tsv_data: bytes,
+) -> None:
+    json_target = require_output_path(json_path)
+    tsv_target = require_output_path(tsv_path)
+    if json_target.parent != tsv_target.parent:
+        raise DryRunOpeningSamplerGateError("paired output paths must share one parent directory")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)